File: people_database/ruby/lib/people_database.rb
require_relative 'person.rb'
require 'pry'

class PeopleDatabase
  attr_accessor :people

  def initialize
    @people = []
  end

  def add(data)
    @people << Person.new(data)
  end

  def remove(email)
    @people.delete_if { |person| person.email == email }
  end

  def all_people(state)
    chosen_people = []
    @people.each { |person| chosen_people << person if person.state == state }
    chosen_people
  end

  def emails
    emails = @people.map { |person| person.email }
    emails.join(", ")
  end

  def state_counts
    state_counts = {}
    @people.each do |person|
      if state_counts[person.state]
        state_counts[person.state] += 1
      else
        state_counts[person.state] = 1
      end
    end
    state_counts
  end

end

File: people_database/js/people_database.js
	var peopleDatabase = function() {
  var Person = require('./person')
  this.people = [];

  this.add = function(person_args) {
    var person = new Person(person_args);
    this.people.push(person);
  };

  this.remove = function(email) {
    var length = this.people.length;
    for (var i = 0; i < length; i++) {
      if (this.people[i].email === email) {
        delete this.people[i];
      };
    };
  };

  this.collect_state = function(state) {
    var length = this.people.length;
    var state_coll = []
    for (var i = 0; i < length; i++) {
      if (this.people[i].state === state) {
        state_coll.push(this.people[i]);
      };
    };
    return state_coll
  };

  this.state_count = function(state) {
    return this.collect_state(state).length;
  };

  this.emails = function() {
    var length = this.people.length;
    var emails = []
    for (var i = 0; i < length; i++) {
      emails.push(this.people[i].email);
    };
    return emails.join(', ')
  };
};

module.exports = peopleDatabase;
